### PR TITLE
feat: Initial parity with latest WinUI Expander

### PR DESCRIPTION
**Pull request by duke7553**
GitHub Issue (If applicable): #6345 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Our Expander implementation has grown out of date with that from WinUI.

## What is the new behavior?
Updated to have the latest resources, markup and general code behind

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
**Please validate these changes carefully as this is my first time reimplementing a control from WinUI.**

Internal Issue (If applicable): N/A

### CODE REVIEW

**Comment by robloo** [comment]
There is already a PR for this #5626.

The reason it wasn't finished at the time is because Visual Studio was causing a lot of build issues. These may have been fixed since then and I was about to take another look https://github.com/unoplatform/uno/issues/5082#issuecomment-867894789. 

How do you want to resolve this? Keep in mind I was also updating ColorPicker.

**Comment by robloo** [comment]
You also didn't indicate in the source code, commits or https://github.com/unoplatform/uno/blob/master/src/Uno.UI/Microsoft/UI/Xaml/WinUIVersionLog.txt which level of WinUI you updated to. This is a problem for future change tracking.

**Comment by duke7553** [comment]
@robloo This PR was marked as a draft. However, I will definitely update the commit string to the latest one.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs [hunk 1]
```diff
@@ -66,10 +70,30 @@ protected override void OnApplyTemplate()
...
 
+			//if (this.GetTemplateChild<Border>(c_expanderContentClip) is Border expanderContentClip)
+			//{
+			//	var visual = ElementCompositionPreview.GetElementVisual(expanderContentClip);
+			//	visual.Clip(visual.Compositor().CreateInsetClip());
+			//}
+
+			if (this.GetTemplateChild<Border>(c_expanderContent) is Border expanderContent)
+			{
+				expanderContent.SizeChanged += OnContentSizeChanged;
```
Comment (robloo) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:81]: In my PR I added a SerialDisposable to handle removing the event handler here. This is important as I understand things for iOS.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs [hunk 2]
```diff
@@ -66,10 +70,30 @@ protected override void OnApplyTemplate()
 				}
 			}
 
+			//if (this.GetTemplateChild<Border>(c_expanderContentClip) is Border expanderContentClip)
```
Comment (robloo) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/Expander.cs:73]: Was this disabled due to know issues? In my PR it is still enabled -- however, I wasn't able to test due to build issues.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/ExpanderTemplateSettings.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{
+	public partial class ExpanderTemplateSettings : DependencyObject
+	{
+		public double ContentHeight
+		{
+			get => (double)GetValue(ContentHeightProperty);
+			set => SetValue(ContentHeightProperty, value);
+		}
+
+		public static DependencyProperty ContentHeightProperty { get; } =
```
Comment (robloo) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/ExpanderTemplateSettings.cs:19]: I know it's more verbose, but I think this should be broken up following conventions in WinUI. There should be separate `ExpanderTemplateSettings.Properties.cs` and `ExpanderTemplateSettings.cs` files as in PR #5626.

**Comment by robloo** [comment]
I think it makes the most sense if we keep #5626 open as it includes ColorPicker changes as well as does some things differently for expander I think we want (keeps files closer to WinUI, cleanly manages event handlers, etc.). I can review this PR side-by-side and copy over relevant changes. XAML, for example, I never properly merged disabling some things Uno doesn't support. The clipping may also need to be disabled as you did.

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/ExpanderTemplateSettings.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+using Windows.UI.Xaml;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class ExpanderTemplateSettings : DependencyObject
+	{
+		public double ContentHeight
+		{
+			get => (double)GetValue(ContentHeightProperty);
+			set => SetValue(ContentHeightProperty, value);
```
Comment (robloo) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/Expander/ExpanderTemplateSettings.cs:16]: These are internal only setters as well:

https://github.com/microsoft/microsoft-ui-xaml/blob/090fd4d11be744473324056d8d00dce797cdda6c/dev/Expander/Expander.idl#L9

**Comment by MartinZikmund** [comment]
@robloo @duke7553 I'm really sorry, I completely forgot there is already a PR that updates the `Expander`, so that's on me 😅 . Are there blockers to merge that one? If not we can probably merge both `Expander` and `ColorPicker` updates at once, as we will need both anyway.

**Comment by robloo** [comment]
No need to apologize! I'll get things updated this week so the other PR is ready to merge.

**Pull request closed without merging**